operators/multiclusterobservability/controllers/placementrule/hub_ocp_monitoring_util.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+// Licensed under the Apache License 2.0
+
+package placementrule
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	cmomanifests "github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	operatorconfig "github.com/stolostron/multicluster-observability-operator/operators/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const ( // #nosec G101 -- Not a hardcoded credential.
+	hubAmRouterCASecretName        = "hub-alertmanager-router-ca"
+	clusterMonitoringConfigName    = "cluster-monitoring-config"
+	clusterMonitoringConfigDataKey = "config.yaml"
+	endpointMonitoringOperatorMgr  = "endpoint-monitoring-operator"
+	promNamespace                  = "openshift-monitoring"
+	clusterRoleBindingName         = "metrics-collector-view"
+	caConfigmapName                = "metrics-collector-serving-certs-ca-bundle"
+	hubMetricsCollectionNamespace  = "open-cluster-management-observability"
+	etcdServiceMonitor             = "acm-etcd"
+	kubeApiServiceMonitor          = "acm-kube-apiserver"
+)
+
+// revertClusterMonitoringConfig reverts the configmap cluster-monitoring-config and relevant resources
+// (observability-alertmanager-accessor and hub-alertmanager-router-ca) for the openshift cluster monitoring stack.
+func RevertHubClusterMonitoringConfig(ctx context.Context, client client.Client) error {
+	log.Info("revertClusterMonitoringConfig called")
+
+	// try to retrieve the current configmap in the cluster
+	found := &corev1.ConfigMap{}
+	err := client.Get(ctx, types.NamespacedName{Name: clusterMonitoringConfigName,
+		Namespace: promNamespace}, found)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("configmap not found, no need action", "name", clusterMonitoringConfigName)
+			return nil
+		} else {
+			log.Error(err, "failed to check configmap", "name", clusterMonitoringConfigName)
+			return err
+		}
+	}
+
+	// do not touch the configmap if are not already a manager
+	touched := false
+	log.Info("checking field.Manager")
```

Comment:
A lot of these logs are debug level to me. I think we are generating too much noise at the info level.